libr/include/r_bin.h

```diff
@@ -224,6 +224,7 @@ typedef struct r_bin_object_t {
 	ut64 size;
 	ut64 obj_size;
 	RList/*<RBinSection>*/ *sections;
+	bool segments_only;
```

Comment:
is it really necessary adding this field in the RBinObject? Can't you compute it in the `bin_sections` function since you are already iterating over all sections? If I understand it correctly, it would just be a matter of checking if there's any section with `is_segment == false`. That way, you don't need to set segments_only in each binplugin.

Reply:
ofc not, yes b89bb4c206f45f1ca619c6c57328d134d8479e49